File: jushi_backend/app/models/book_analysis.py
"""
书籍 PDF -> NotebookLM 分析相关数据模型
"""

from __future__ import annotations

from typing import List, Literal, Optional

from pydantic import BaseModel, Field


ProjectStatus = Literal["draft", "running", "completed", "failed", "completed_with_errors"]
ChapterStatus = Literal["draft", "running", "completed", "failed", "completed_with_errors"]


class BookAnalysisChapterInput(BaseModel):
    title: str = Field(..., min_length=1, max_length=300)
    startPage: int = Field(..., ge=1)
    endPage: int = Field(..., ge=1)


class BookAnalysisChapterUpdateRequest(BaseModel):
    chapters: List[BookAnalysisChapterInput] = Field(..., min_length=1)


class BookAnalysisChapter(BaseModel):
    id: str
    title: str
    startPage: int
    endPage: int
    status: ChapterStatus = "draft"
    summary: str = ""
    keyPoints: List[str] = Field(default_factory=list)
    arguments: List[str] = Field(default_factory=list)
    examples: List[str] = Field(default_factory=list)
    evidence: List[str] = Field(default_factory=list)
    quotedEvidence: List[str] = Field(default_factory=list)
    openQuestions: List[str] = Field(default_factory=list)
    rawAnswer: str = ""
    error: Optional[str] = None


class BookAnalysisCreateResponse(BaseModel):
    project: dict

File: mobile/jushi_mobile/types/book-analysis.ts
export type BookAnalysisStatus =
  | 'draft'
  | 'running'
  | 'completed'
  | 'failed'
  | 'completed_with_errors';

export interface BookAnalysisChapter {
  id: string;
  title: string;
  startPage: number;
  endPage: number;
  status: BookAnalysisStatus;
  summary: string;
  keyPoints: string[];
  arguments: string[];
  examples: string[];
  evidence: string[];
  quotedEvidence: string[];
  openQuestions: string[];
  rawAnswer: string;
  error: string | null;
}

export interface BookAnalysisProject {
  id: string;
  userId: string;
  title: string;
  originalFilename: string;
  sourcePath: string;
  sourceRawUrl?: string;
  pageCount: number;
  notebookId?: string | null;
  status: BookAnalysisStatus;
  currentStage: string;
  currentChapterIndex: number;
  totalChapters: number;
  progressPercent: number;
  error?: string | null;
  chapters: BookAnalysisChapter[];
  exportHtmlPath?: string | null;
  exportHtmlUrl?: string | null;
  createdAt?: string;
  updatedAt?: string;
  startedAt?: string | null;
  completedAt?: string | null;
}
